src/components/AddToBasketButton/AddToBasketButton.js

```diff
@@ -74,24 +78,62 @@ class AddToBasketButton extends React.Component {
 
   render() {
     const { itemExistsInBasket } = this.state;
-    const { addLabel, buttonProps, disabled, item, removeLabel } = this.props;
+    const {
+      addButtonTooltipText,
+      addLabel,
+      buttonProps,
+      disabled,
+      item,
+      removeButtonTooltipText,
+      removeLabel,
+    } = this.props;
+
     const coverage = JSON.stringify(get(item, '_object.coverage[0]', {}));
 
     return (
-      <IfPermission perm="ui-agreements.agreements.edit">
-        <Button
-          {...buttonProps}
-          buttonStyle={itemExistsInBasket ? 'default' : 'primary'}
-          data-test-basket-add-button={itemExistsInBasket ? undefined : true}
-          data-test-basket-remove-button={itemExistsInBasket ? true : undefined}
-          data-test-coverage-details={coverage}
-          data-test-entitlement-option-id={item.id}
-          disabled={disabled}
-          onClick={itemExistsInBasket ? this.removeFromBasket : this.addToBasket}
-        >
-          {itemExistsInBasket ? removeLabel : addLabel}
-        </Button>
-      </IfPermission>
+      addButtonTooltipText ?
+        (
+          <Tooltip
+            id={uniqueId('addtobasketbutton')}
+            text={itemExistsInBasket ? removeButtonTooltipText : addButtonTooltipText}
+          >
+            {({ ref, ariaIds }) => (
+              <IfPermission perm="ui-agreements.agreements.edit">
+                <Button
+                  {...buttonProps}
+                  aria-labelledby={ariaIds.text}
+                  buttonStyle={itemExistsInBasket ? 'default' : 'primary'}
+                  data-test-basket-add-button={itemExistsInBasket ? undefined : true}
+                  data-test-basket-remove-button={itemExistsInBasket ? true : undefined}
+                  data-test-coverage-details={coverage}
```

Comment:
Can we clean up all these props that are duplicated across the two render modes (w/ Tooltip and w/o)? Ie, create a `baseProps` variable and spread that object into the `Button`'s props.

(I just merged this [eslint-config-stripes PR](https://github.com/folio-org/eslint-config-stripes/pull/54/files) so if you re-yarn in the new build, you won't need to quote all the props, just the ones with hyphens)

Reply:
Also, I think you can wrap the entire thing in an `IfPermission` instead, so you don't have to define that twice here as well.
```
const button = addButtonTooltipText ? <Tooltip /> : <Button />;
return <IfPermission>{button}</IfPermission>;
```